scripts/generating_yaml.py

```diff
@@ -0,0 +1,213 @@
+
+# coding: utf-8
+
+# In[1]:
+
+from __future__ import print_function
+import yaml
+from ipywidgets import interact, interactive, fixed, interact_manual
+import ipywidgets as widgets
+from IPython.display import display
+import os
+import datetime
+
+
+# In[2]:
+
+device_data = {'serial_into_USBhub_port' : [0, 1, 2, 3, 4, 5, 6, 7],
+               'camera_into_serial_port' : [1, 2, 3, 4],
+               'USBhub_SN' : ['OX518EFFE1', 'OXCD12637D'],
+               'camera_into_USBhub_port' : [0, 1, 2, 3, 4, 5, 6, 7],
+               'sbig_SN' : ['83F011167', '83F011791', '83F011639', '83F010801', '83F010774', '83F011758', '83F010771', '83F011810'],
+              'birger_SN': ['10858', '14287', '14286', '14285', '13281', '13134', '14284', '13208', '14276'],
+              'lens_SN': ['3360000099', '3360000063', '3360000087', '2850000067', '3150000110', '5370000054'],
+              'filter_ID': ['r2_1', 'r2_2', 'g2_3', 'g2_4', 'r2_5', 'r2_6', 'r2_7', 'g2_8', 'g2_9', 'ha1_10', 'ha1_11', 'g2_12', 'ha1_13', 'ha1_14']
+              }
+
+
+# In[3]:
+
+with open('serial_management.yaml', 'w') as file:
+    yaml.dump(device_data)
+    file.write(yaml.dump(device_data))
+
+with open('serial_management.yaml', 'r') as file:
+    data = yaml.load(file)
+
+
+# In[4]:
+
+def type_filename(type_filename_here):
+    return type_filename_here
+
+
+# In[5]:
+
+file_name = interactive(type_filename, type_filename_here="filename.yaml")
+
+
+# In[6]:
+
+data_list = {'name' : 'HuntsmanSSO',
+             'log_file' : 'huntsman',
+             'location' : {'name' : 'Siding Spring Observatory',
+                           'latitude' : -31.27, #degrees
+                           'longitude': 149.06, # Degrees
+                          'elevation': 1000.0, # Meters
```

Comment:
Just adding this for reference like we discussed, the data list might be able to be imported straight from the config file.